### PR TITLE
[Caffe2] Add full impl of GroupNorm

### PR DESCRIPTION
**Pull request by xiaomengy**

### CODE REVIEW

File: caffe2/operators/group_norm_op.cc [hunk 1]
```diff
@@ -1,16 +1,196 @@
...
 
+#include <array>
+
+#include "caffe2/utils/math.h"
+
 namespace caffe2 {
 
+namespace {
+
+template <typename T>
```
Comment (houseroad) [inline caffe2/operators/group_norm_op.cc]: Shall we lift this to math.h?

File: caffe2/operators/group_norm_op.cc [hunk 2]
```diff
@@ -1,16 +1,196 @@
...
+inline T Cube(const T& x) {
```
Comment (houseroad) [inline caffe2/operators/group_norm_op.cc]: Oh, it's inline, probably no need.

Comment (BIT-silence) [reply caffe2/operators/group_norm_op.cc]: Yes, and this one is a very simple function and Cube is not very popular in our use case. So I think there is no need to move it.

**Review by houseroad — APPROVED** [review]
LGTM, thanks!

**Pull request merged**